jvm-packages/xgboost4j-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostGeneralSuite.scala

```diff
@@ -367,4 +367,37 @@ class XGBoostGeneralSuite extends FunSuite with TmpFolderPerSuite with PerTest {
     df2.collect()
   }
 
+  test("Getter and setter for AllowNonZeroForMissingValue works") {
+    {
+      val paramMap = Map("eta" -> "1", "max_depth" -> "6",
+        "objective" -> "binary:logistic", "num_round" -> 5, "num_workers" -> numWorkers)
+      val training = buildDataFrame(Classification.train)
+      val classifier = new XGBoostClassifier(paramMap)
+      classifier.setAllowNonZeroForMissing(true)
+      assert(classifier.getAllowNonZeroForMissingValue)
+      classifier.setAllowNonZeroForMissing(false)
+      assert(!classifier.getAllowNonZeroForMissingValue)
+      val model = classifier.fit(training)
+      model.setAllowNonZeroForMissing(true)
+      assert(model.getAllowNonZeroForMissingValue)
+      model.setAllowNonZeroForMissing(false)
+      assert(!model.getAllowNonZeroForMissingValue)
```

Comment:
can we add in MissValueSuite

Reply:
So move this unit test to the new Suite?

Reply:
No, there is a MissingValueHandlingSuite.scala, but I am not sure if we want to test a getter/setter, if we look at the last unit test there, it has tested the behavior of this parameter

Reply:
Does `allow_non_zero_for_missing` parameter in the param map have the same effect as calling `setAllowNonZeroForMissing()`?

Reply:
yes, it is equivalet

Reply:
Got it. I think we should have a small test for getter and setter still, because previously the getter and setter were misspelled and thus broken.

Reply:
The test is now moved to MissingValueHandlingSuite.